### PR TITLE
feat: swap 'mouse' events to 'pointer' events for mobile compatibility; update site and docs

### PR DESCRIPTION
**Pull request by alliefeldman**
- Site examples now use ‘pointer’ events instead of mouse events; all other elements of examples remain unchanged
- Update ‘site/docs/’ to include ‘pointer’ events
- All site examples use ‘pointer’ instead of ‘mouse’ in the instances of -over, -move, -out, -down, -up
- Unit tests include pointer events
- All lint and unit tests pass

### CODE REVIEW

**Comment by domoritz** [comment]
Can you add feat: to the title?

**Comment by alliefeldman** [comment]
Done!

File: site/_data/versions.yml
Comment (domoritz) [inline site/_data/versions.yml:1]: Please don't add this file

File: site/.ruby-version
Comment (domoritz) [inline site/.ruby-version:1]: please remove

File: site/=
Comment (domoritz) [inline site/=:1]: please remove

File: site/Gemfile.lock
Comment (domoritz) [inline site/Gemfile.lock:1]: please remove

File: site/docs/parameter/select.md [hunk 1]
```diff
@@ -82,7 +82,7 @@ The `on` property modifies the event that triggers the selection.
 
 For instance, a single rectangle in the heatmap below can now be selected on mouse hover instead.
 
-<div class="vl-example" data-name="selection_type_single_mouseover"></div>
+<div class="vl-example" data-name="selection_type_single_pointerover"></div>
```
Comment (domoritz) [inline site/docs/parameter/select.md:85]: Please make sure you rename the file as well.

File: src/compile/selection/interval.ts [hunk 2]
```diff
@@ -106,13 +106,13 @@ const interval: SelectionCompiler<'interval'> = {
...
         ...(dataSignals.length
           ? {
-              on: [
-                {
-                  events: [{signal: dataSignals.join(' || ')}], // Prevents double invocation, see https://github.com/vega/vega/issues/1672.
-                  update: `${dataSignals.join(' && ')} ? {${update}: [${dataSignals}]} : null`
-                }
-              ]
-            }
+            on: [
```
Comment (domoritz) [inline src/compile/selection/interval.ts:109]: Please fix formatting issues.

File: src/compile/selection/interval.ts [hunk 3]
```diff
@@ -233,7 +233,7 @@ const interval: SelectionCompiler<'interval'> = {
 
     // Two brush marks ensure that fill colors and other aesthetic choices do
     // not interefere with the core marks, but that the brushed region can still
-    // be interacted with (e.g., dragging it around).
+    // be interacted with (e.g., ging it around).
```
Comment (domoritz) [inline src/compile/selection/interval.ts:236]: Fix

File: test-runtime/index.html [hunk 4]
```diff
@@ -1,111 +1,114 @@
 <!DOCTYPE html>
 <html>
```
Comment (domoritz) [inline test-runtime/index.html:2]: Why is this file changing so much?

**Review by domoritz — CHANGES_REQUESTED** [review]
Please clean up the pull request. Otherwise looks good.

File: Icon
Comment (domoritz) [inline Icon:1]: Please remove this file. How did this even get here? Be careful when you add files.

File: build/vega-lite-schema.json [hunk 5]
```diff
@@ -19309,6 +19329,9 @@
           ],
           "description": "Clears the selection, emptying it of all values. This property can be a [Event Stream](https://vega.github.io/vega/docs/event-streams/) or `false` to disable clear.\n\n__Default value:__ `dblclick`.\n\n__See also:__ [`clear` examples ](https://vega.github.io/vega-lite/docs/selection.html#clear) in the documentation."
         },
+        "cursor": {
```
Comment (domoritz) [inline build/vega-lite-schema.json:19332]: Odd. Why does this happen in your pr?

File: examples/specs/concat_hover_filter.vl.json
Comment (domoritz) [inline examples/specs/concat_hover_filter.vl.json:1]: Please don't change the formatting of existing files. It was often intentional.

File: site/Gemfile.lock
Comment (domoritz) [inline site/Gemfile.lock:1]: Don't delete the lockfile

File: site/=
Comment (domoritz) [inline site/=:1]: Delete this

File: src/channel.ts [hunk 6]
```diff
@@ -72,6 +72,8 @@ export const HREF = 'href' as const;
 export const URL = 'url' as const;
 export const DESCRIPTION = 'description' as const;
 
+// export const CURSOR = 'cursor' as const;
```
Comment (domoritz) [inline src/channel.ts:75]: Remove this and look craefully through your changes.

File: src/channeldef.ts [hunk 7]
```diff
@@ -1028,6 +1027,8 @@ export function defaultType<T extends TypedFieldDef<Field>>(fieldDef: T, channel
   return 'nominal';
 }
 
+// export type cursor = Cursor;
```
Comment (domoritz) [inline src/channeldef.ts:1030]: same

File: src/channeldef.ts [hunk 8]
```diff
@@ -178,7 +178,6 @@ export interface ConditionValueDefMixins<V extends Value = Value> {
  *   ...
  * }
  */
```
Comment (domoritz) [inline src/channeldef.ts:180]: why delete a line?

File: src/encoding.ts [hunk 9]
```diff
@@ -52,6 +52,7 @@ import {
   binRequiresRange,
   ChannelDef,
   ColorDef,
+  // Cursor,
```
Comment (domoritz) [inline src/encoding.ts:55]: Don't make these changes here

File: src/selection.ts [hunk 10]
```diff
@@ -162,18 +164,30 @@ export interface BrushConfig {
 }
 
 export interface IntervalSelectionConfig extends BaseSelectionConfig<'interval'> {
+  /**
```
Comment (domoritz) [inline src/selection.ts:167]: Don't make multiple changes in one pr.

File: test-runtime/index.html (hunk 4 above)
Comment (domoritz) [reply test-runtime/index.html:2]: It seems like you have some kind of formatter that runs automatically. Please try not to make big changes like formatting changes in a pr. Keep the diff minimal.

**Review by domoritz — CHANGES_REQUESTED** [review]
Please look over the pull request more carefully. There are a lot of cahnges that should not be included.

**Review by domoritz — CHANGES_REQUESTED** [review]
Please look over the pull request more carefully. There are a lot of cahnges that should not be included.

File: test-runtime/index.html (hunk 4 above)
Comment (alliefeldman) [reply test-runtime/index.html:2]: I've reset all of my formatting settings, which have resolved a good number of the formatting issues in the compiled examples. However, the "prettier" formatter used seems to be stubborn about specifically the spacing between `<html>` and `<head>`, which is changing the starting lines for all of the following code segments.

File: site/Gemfile.lock
Comment (domoritz) [inline site/Gemfile.lock:1]: Please add this file back

File: test-runtime/index.html (hunk 4 above)
Comment (domoritz) [reply test-runtime/index.html:2]: I see. Ideally you would not include this in the pull request but I guess if prettier insists, we can accept this change now. Usually I'd like these formatting changes in a separate pull request.

File: examples/specs/interactive_line_point_hover.vl.json [hunk 11]
```diff
@@ -8,22 +8,22 @@
...
-    "mark": {"type": "circle", "tooltip": true},
-    "encoding": {
-      "opacity": {
-        "condition": {"test": {"param": "hover", "empty": false}, "value": 1},
-        "value": 0
-      },
-      "size": {
-        "condition": {"test": {"param": "hover", "empty": false}, "value": 48},
-        "value": 100
+      "mark": "line"
```
Comment (domoritz) [inline examples/specs/interactive_line_point_hover.vl.json:11]: Let's leave the indentation as it was before.

**Review by domoritz — CHANGES_REQUESTED** [review]
Delete the versions file as well.

File: .gitignore [hunk 12]
```diff
@@ -18,4 +18,4 @@ site/static/bundle.*
 test/log
 test/log/difflist.json
 yarn-debug.log*
-yarn-error.log*
+yarn-error.log*
```
Comment (domoritz) [inline .gitignore:21]:
```suggestion
yarn-error.log*

```

**Review by domoritz — APPROVED** [review]

**Pull request merged**